core/src/main/java/org/infinispan/metrics/impl/MetricsCollector.java

```diff
@@ -57,27 +59,51 @@
    @Inject
    ComponentRef<Transport> transportRef;
 
-   protected MetricsCollector(MetricRegistry registry) {
-      this.registry = registry;
+   protected MetricsCollector() {
+      this.baseRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+      this.vendorRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+   }
+
+   public PrometheusMeterRegistry getBaseRegistry() {
+      return baseRegistry;
    }
 
-   public MetricRegistry getRegistry() {
-      return registry;
+   public PrometheusMeterRegistry getVendorRegistry() {
+      return vendorRegistry;
    }
 
    @Start
    protected void start() {
+      baseRegistry.config().meterFilter(new BaseFilter());
+      new BaseAdditionalMetrics().bindTo(baseRegistry);
+
+      vendorRegistry.config().meterFilter(new VendorFilter());
+      new VendorAdditionalMetrics().bindTo(vendorRegistry);
+
       Transport transport = transportRef.running();
       String nodeName = transport != null ? transport.getAddress().toString() : globalConfig.transport().nodeName();
       if (nodeName == null) {
          //TODO [anistor] Maybe we should just ensure a unique node name was set in all tests and also in real life usage, even for local cache managers
          nodeName = generateRandomName();
          //throw new CacheConfigurationException("Node name must always be specified in configuration if metrics are enabled.");
       }
-      nodeTag = new Tag(NODE_TAG_NAME, nodeName);
+      nodeTag = Tag.of(NODE_TAG_NAME, nodeName);
 
       if (globalConfig.metrics().namesAsTags()) {
-         cacheManagerTag = new Tag(CACHE_MANAGER_TAG_NAME, globalConfig.cacheManagerName());
+         cacheManagerTag = Tag.of(CACHE_MANAGER_TAG_NAME, globalConfig.cacheManagerName());
+      }
+   }
+
+   @Stop
+   protected void stop() {
+      try {
+         if (baseRegistry != null) {
```

Comment:
This can never be true as these are initialized in the constructor. Maybe instead we should initialize in start and assign to null in this if block? This way the component is also completely reusable.

Reply:
Right. Done